samples/azure-quantum/utilities/storage-data-management.ipynb

```diff
@@ -0,0 +1,758 @@
+      "metadata": {
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "# Data Management Sample\n",
+        "This sample shows you how to connect and use Azure Quantum with external datasources such as Azure Blob Storage. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "## Work with data in Azure Blob Storage\n",
+        "This section will show you how to work with JSON and other file types in Blob Storage.\n",
+        "\n",
+        "Prerequisites:\n",
+        "- Storage Account\n",
+        "\n",
+        "You must have a Azure Storage account deployed to use Blob Storage for your data. If you are accessing this sample through Azure Quantum hosted notebooks, you should already have a Storage account set up for your [Quantum Workspace](https://docs.microsoft.com/azure/quantum/how-to-create-workspace?tabs=tabid-quick). \n",
+        "You can find details for this storage account by navigating to your Azure Quantum Workspace in the portal - it is shown in the 'Essentials' section at the top - if you click the name of the storage account it will take you to view it in the portal.\n",
+        "\n",
+        "If you wish to use a different storage account, you can absolutely do so. More information on setting up a storage account can be found [here](https://docs.microsoft.com/azure/storage/common/storage-account-create?tabs=azure-portal).\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "source": [
+        "### Setup\n",
+        "You can find more information about the Blob Storage SDK [here](https://docs.microsoft.com/python/api/overview/azure/storage-blob-readme?view=azure-python)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "jupyter": {
+          "outputs_hidden": false,
+          "source_hidden": false
+        },
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "outputs": [],
+      "source": [
+        "# Import Azure Storage Python SDK & json utils\n",
+        "from azure.storage.blob import BlobServiceClient\n",
+        "import json"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "jupyter": {
+          "outputs_hidden": false,
+          "source_hidden": false
+        },
+        "nteract": {
+          "transient": {
+            "deleting": false
+          }
+        }
+      },
+      "outputs": [],
+      "source": [
+        "# Log in to the Azure CLI (only need to do once per session)\n",
+        "!az login"
```

Comment:
@anpaz, is there any way to use the existing azure-quantum libs to connect to the linked storage account without doing the !az login method in this sample?

I found this way to get a blob, but it seems a bit hacky:

from azure.quantum import Workspace
workspace = Workspace (
    subscription_id = "677fc922-91d0-4bf6-9b06-4274d319a0fa",
    resource_group = "xiou",
    name = "xiou-notebooks-demo",
    location = "eastus2euap"
)

from azure.quantum._client.models import BlobDetails
sas_uri_response = workspace._client.storage.sas_uri(blob_details=BlobDetails(container_name="my-container"))

from azure.storage.blob import ContainerClient
container_client = ContainerClient.from_container_url(container_url=sas_uri_response.sas_uri)
container_client.exists()